src/databricks/labs/ucx/source_code/notebook.py

```diff
@@ -232,37 +241,49 @@ def new_cell(self, source: str) -> Cell:
     def extract_cells(self, source: str) -> list[Cell] | None:
         lines = source.split('\n')
         header = f"{self.comment_prefix} {NOTEBOOK_HEADER}"
+
         if not lines[0].startswith(header):
             raise ValueError("Not a Databricks notebook source!")
 
-        def make_cell(cell_lines: list[str]):
+        def make_cell(cell_lines: list[str], start: int = 0):
             # trim leading blank lines
             while len(cell_lines) > 0 and len(cell_lines[0]) == 0:
                 cell_lines.pop(0)
+                start += 1
+
             # trim trailing blank lines
             while len(cell_lines) > 0 and len(cell_lines[-1]) == 0:
                 cell_lines.pop(-1)
             cell_language = self.read_cell_language(cell_lines)
             if cell_language is None:
                 cell_language = self
             else:
+                before = len(cell_lines)
                 self._remove_magic_wrapper(cell_lines, cell_language)
+                start += before - len(cell_lines)
+
             cell_source = '\n'.join(cell_lines)
-            return cell_language.new_cell(cell_source)
+            cell = cell_language.new_cell(cell_source)
```

Comment:
Yep, and probably you'll have to change a signature of this method as well to support constructor injection

Reply:
Signature changed